packages/integration/src/lib/authentication/auth-state.ts

```diff
@@ -324,13 +329,22 @@ export async function getAllWalletsFromThisDevice() {
       return deserializeUserIdData(wallet as string)
     })
     .filter((profile) => profile.cacheVersion === EXPECTED_CACHE_VERSION)
-  const profilesData = profiles.map((profile) => {
-    return {
-      email: profile.email,
-      principal: profile.userId,
+
+  const profilesData = profiles.reduce((acc, profile) => {
+    const newProfile = {
+      email: profile.email ?? "",
```

Comment:
please don't use empty strings. I think in ts there are 2 options: null or undefined

Reply:
and I guess it would good to filter out profiles without emails. I don't think that we have such accounts but who knows....